protocol/app/app.go

```diff
@@ -1222,6 +1230,31 @@ func New(
 	return app
 }
 
+// MonitorDaemon registers a daemon service with the update monitor. If the daemon does not register, the method will
+// panic.
+func (app *App) MonitorDaemon(
```

Comment:
```suggestion
func (app *App) RegisterDaemonHealthMonitor(
```

Reply:
I changed the method name slightly to `RegisterDaemonWithHealthMonitor` because the above rename seemed easy to mis-parse, i.e. we are not registering a daemon health monitor, but registering a daemon with the health monitor.